src/main/java/org/opensearch/search/relevance/transformer/personalizeintelligentranking/client/PersonalizeClientSettings.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.search.relevance.transformer.personalizeintelligentranking.client;
+
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.auth.BasicSessionCredentials;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.settings.SecureString;
+import org.opensearch.common.settings.SecureSetting;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.settings.SettingsException;
+
+/**
+ * Container for personalize client settings such as AWS credentials
+ */
+public final class PersonalizeClientSettings {
+
+    private static final Logger logger = LogManager.getLogger(PersonalizeClientSettings.class);
+
+    /**
+     * The access key (ie login id) for connecting to Personalize.
+     */
+    public static final Setting<SecureString> ACCESS_KEY_SETTING = SecureSetting.secureString("personalize_intelligent_ranking.aws.access_key", null);
+
+    /**
+     * The secret key (ie password) for connecting to Personalize.
+     */
+    public static final Setting<SecureString> SECRET_KEY_SETTING = SecureSetting.secureString("personalize_intelligent_ranking.aws.secret_key", null);
+
+    /**
+     * The session token for connecting to Personalize.
+     */
+    public static final Setting<SecureString> SESSION_TOKEN_SETTING = SecureSetting.secureString("personalize_intelligent_ranking.aws.session_token", null);
+
+    private final AWSCredentials credentials;
+
+    protected PersonalizeClientSettings(AWSCredentials credentials) {
+        this.credentials = credentials;
+    }
+
+    public AWSCredentials getCredentials() {
+        return credentials;
+    }
+
+    /**
+     * Load AWS credentials from open search keystore if available
+     * @param settings Open search settings
+     * @return AWS credentials
+     */
+    static AWSCredentials loadCredentials(Settings settings) {
+        try (SecureString key = ACCESS_KEY_SETTING.get(settings);
+             SecureString secret = SECRET_KEY_SETTING.get(settings);
+             SecureString sessionToken = SESSION_TOKEN_SETTING.get(settings)) {
+            if (key.length() == 0 && secret.length() == 0) {
+                if (sessionToken.length() > 0) {
+                    throw new SettingsException("Setting [{}] is set but [{}] and [{}] are not",
+                            SESSION_TOKEN_SETTING.getKey(), ACCESS_KEY_SETTING.getKey(), SECRET_KEY_SETTING.getKey());
+                }
+                logger.info("Using either environment variables, system properties or instance profile credentials");
+                return null;
```

Comment:
Should this return a `DefaultAWSCredentialsProviderChain`? 

I guess you would need the other branch to return an instance of AWSCredentialsProvider too, but you could just wrap the return value in an instance of `AWSStaticCredentialsProvider`.

Reply:
This method is used for getting credentials from open search keystore. There is another class PersonalizeCredentailsProviderFactory which implements logic for providing appropriate implementation of AWSCredentialsProvider based on availability of different credentials options.